src/DataModel.cpp

```diff
@@ -1,16 +1,83 @@
 #include <Zivid/DataModel/NodeType.h>
+#include <Zivid/Point.h>
+#include <Zivid/Range.h>
 
 #include <ZividPython/DataModel.h>
 #include <ZividPython/NodeType.h>
 #include <ZividPython/Wrappers.h>
 
+#include "pybind11/numpy.h"
+#include <pybind11/operators.h>
 #include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+namespace ZividPython
+{
+    void wrapClass(pybind11::class_<Zivid::Range<double>> pyClass)
+    {
+        using R = Zivid::Range<double>;
+        pyClass.def(pybind11::init<>())
+            .def(pybind11::init([](double min, double max) {
+                return R{ min, max };
+            }))
+            .def(pybind11::init([](const std::array<double, 2> &array) {
+                return R{ array[0], array[1] };
+            }))
+            .def_property_readonly("min", [](const R &r) { return r.min(); })
+            .def_property_readonly("max", [](const R &r) { return r.max(); })
+            .def("to_array",
+                 [](const R &self) {
+                     return std::array<double, 2>{ self.min(), self.max() };
+                 })
+            .def("__repr__", &R::toString)
+            .def("to_string", &R::toString)
+            .def(pybind11::self == pybind11::self) // NOLINT
+            .def(pybind11::self != pybind11::self) // NOLINT
+            .def("is_in_range", &R::isInRange);
+
+        pybind11::implicitly_convertible<pybind11::tuple, R>();
```

Comment:
I tried and failed getting `pybind11::implicitly_convertible<pybind11::iterable, R>();` to work. If this works, it would cover all three and more (like numpy arrays). Any idea why?

Reply:
I tried that here and it seems to work (it builds and all of the tests pass). Pushed fixup now.

Regarding why you failed to do something similar elsewhere: Maybe you were missing some includes? `#include <pybind11/pybind11.h>` is not always sufficient.

Reply:
Nice! Not sure what I messed up, probably includes like you said. I don't think I included anything extra.